pkg/cmd/cluster/cluster_link.go

```diff
@@ -0,0 +1,133 @@
+package cluster
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/url"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/dcos/dcos-cli/pkg/login"
+	"github.com/dcos/dcos-cli/pkg/setup"
+	"github.com/spf13/cobra"
+)
+
+// newCmdClusterLink links the attached cluster to another one.
+func newCmdClusterLink(ctx api.Context) *cobra.Command {
+	setupFlags := setup.NewFlags(ctx.Fs(), ctx.EnvLookup)
+	cmd := &cobra.Command{
+		Use:  "link",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			attachedCluster, err := ctx.Cluster()
+			if err != nil {
+				return err
+			}
+
+			var linkableCluster config.Cluster
+			for _, cluster := range ctx.Clusters() {
+				if args[0] == cluster.ID() ||
+					args[0] == cluster.URL() ||
+					args[0] == cluster.Name() && ctx.IsUniqueCluster(cluster.Name()) {
+					linkableCluster = *cluster
+					break
+				}
+			}
+
+			if (config.Cluster{}) == linkableCluster {
+				// The cluster does not exist yet. Two possibilities:
+				// - The argument is an URL, we try to setup the cluster.
+				// - The argument is not an URL, we return an error.
+				_, err = url.ParseRequestURI(args[0])
+				if err != nil {
+					return errors.New("unable to retrieve cluster " + args[0])
+				}
+
+				msg := " is not set up in the CLI, would you like to do it now?"
+				err = ctx.Prompt().Confirm(args[0] + msg)
+				if err != nil {
+					return err
+				}
+
+				// We save the current configuration before starting the setup.
+				manager := ctx.ConfigManager()
+				conf, err := manager.Current()
+				if err != nil {
+					return err
+				}
+				ctx.Setup(setupFlags, args[0])
```

Comment:
We should check for a possible error here. Also, with this new use-case, how about not attaching the cluster at the end of the setup, but let the caller do it? This would remove the need for saving the current config before the setup, then getting the newly configured cluster and eventually attaching back to the previous one.

So `ctx.Setup()` would return a `(*Config, err)` or `(*Cluster, err)`.